pytket/qir/conversion/qirgenerator.py

```diff
@@ -731,6 +791,153 @@ def conv_classicalexpbox(self, op: ClassicalExpBox, args: list) -> None:
         else:
             self.set_ssa_vars(outputs, output_instruction, True)
 
+    def _get_ssa_from_clexpr_arg(
+        self,
+        arg: int | ClBitVar | ClRegVar | ClExpr,
+        bit_posn: dict[int, int],
+        reg_posn: dict[int, list[int]],
+        cmd_args: list[Bit],
+        expect_register: bool,
+    ) -> tuple[bool, Value]:
+        if isinstance(arg, int):
+            if expect_register:
+                return False, pyqir.const(self.qir_int_type, arg)
+            elif arg not in (0, 1):
+                raise ValueError(f"Invalid bit value {arg}")
+            else:
+                return True, pyqir.const(self.qir_bool_type, arg)
+        elif isinstance(arg, ClBitVar):
+            cmd_arg: Bit = cmd_args[bit_posn[arg.index]]
+            return True, self._get_bit_from_creg(cmd_arg.reg_name, cmd_arg.index[0])
+        elif isinstance(arg, ClRegVar):
+            return False, self._get_i64_ssa_reg(
+                cmd_args[reg_posn[arg.index][0]].reg_name
+            )
+        else:
+            assert isinstance(arg, ClExpr)
+            expr_op: ClOp = arg.op
+            ssa_args = [
+                self._get_ssa_from_clexpr_arg(
+                    expr_arg,
+                    bit_posn,
+                    reg_posn,
+                    cmd_args,
+                    expr_op in _TK_CLEXPR_OP_WITH_REG_ARGS,
+                )[1]
+                for expr_arg in arg.args
+            ]
+            match expr_op:
+                case ClOp.BitAnd | ClOp.BitOr | ClOp.BitXor | ClOp.BitEq | ClOp.BitNeq:
+                    return True, _TK_CLEXPR_OP_TO_PYQIR[expr_op](self.module.builder)(
+                        *ssa_args
+                    )
+                case ClOp.BitNot:
+                    # Implemented as x --> 1 - x
+                    assert len(ssa_args) == 1
+                    return True, self.module.builder.sub(
+                        pyqir.const(self.qir_bool_type, 1), ssa_args[0]
```

Comment:
I think we should put in a request for pyqir to support not operation...
Happy to do that.